frontend/hub/administration/repositories/RepositoryForm.tsx

```diff
@@ -0,0 +1,309 @@
+  pulp_labels: Record<string, string>;
+}
+
+export function RepositoryForm() {
+  const { id } = useParams();
+  const { t } = useTranslation();
+  const pageNavigate = usePageNavigate();
+  const navigate = useNavigate();
+  const getPageUrl = useGetPageUrl();
+  const isEdit = !!id;
+  let isDistributionDisabled = false;
+  let repo: Repository | null;
+  const onSubmit = (data: RepositoryFormProps) => {
+    // format inputs to correct payload
+    const payload = { ...data };
+    if (payload.description === '') {
+      payload.description = null;
+    }
+    delete payload.pipeline;
+    delete payload.hide_from_search;
+    delete payload.createDistribution;
+    payload.pulp_labels = {};
+    if (data.hide_from_search) {
+      payload.pulp_labels.hide_from_search = '';
+    }
+    if (data.pipeline && data.pipeline !== 'none') {
+      payload.pulp_labels.pipeline = data.pipeline;
+    }
+    if (data.remote) {
+      const remote = data.remote as IRemotes;
+      payload.remote = remote.pulp_href;
+    }
+
+    const getDistribution = (repositoryPulpHref: string) => {
+      const basePathTransform = (name: string) => name.replaceAll(/[^-a-zA-Z0-9_/]/g, '_');
+      let distributionName: string = data?.name || '';
+      return postHubRequest(pulpAPI`/distributions/ansible/ansible/`, {
+        base_path: basePathTransform(distributionName),
+        name: data.name,
+        repository: repositoryPulpHref,
+      }).catch(() => {
+        // if distribution already exists, try a numeric suffix to name & base_path
+        distributionName = data.name || '' + Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
+        return postHubRequest(pulpAPI`/distributions/ansible/ansible/`, {
+          base_path: basePathTransform(distributionName),
+          name: data.name,
+          repository: repositoryPulpHref,
+        });
+      });
+    };
+
+    const promise = isEdit
+      ? putHubRequest(
+          pulpAPI`/repositories/ansible/ansible/${
+            parsePulpIDFromURL(repo?.pulp_href || '') || ''
+          }/`,
+          payload
+        )
+      : postRequest<Repository>(pulpAPI`/repositories/ansible/ansible/`, payload);
+    if (data.createDistribution) {
+      if (isEdit) {
+        promise
+          .then(() => {
+            return getDistribution(repo?.pulp_href || '');
+          })
+          .catch(() => t('Distribution not created.'));
+      } else {
+        promise
+          // @ts-expect-error the correct response from promise is known based on the condition
+          .then((repository: Repository) => {
+            return getDistribution(repository?.pulp_href || '');
+          })
+          .catch(() => t('Distribution not created.'));
+      }
+    }
+    return promise
+      .then(() => pageNavigate(HubRoute.RepositoryPage, { params: { id: data.name as string } }))
+      .catch(() => t('The changes not saved.'));
+  };
+  const { data } = useGet<PulpItemsResponse<Repository>>(
```

Comment:
Do we need to get the loading, error, and refresh state from `useGet`?

Reply:
Added.